### ISSUE TITLE
Make outputs available to subsequent steps in the pipeline

### ISSUE THREAD

**Issue by charleszipp** [charleszipp/azure-pipelines-tasks-terraform#120]
## Description
This issue aims to make output variables available to subsequent steps in the pipeline. This is needed when subsequent tasks are dependent on resources provisioned by terraform when those resources' names are non-deterministic.

## Justification
When terraform provisions key vault with an non-deterministic name, the user is unable to connect the key vault task to access secrets also provisioned by terraform. Making output variables available to subsequent steps will unblock access to the key vault secrets provisioned by terraform.

## Solution
When the output command runs, it creates pipeline variables from the output variables produced by the templates. Assuming we have the following outputs defined in the template...

output "lb_address" {
  value = "${aws_alb.web.public_dns}"
}

output "instance_ips" {
  value = ["${aws_instance.web.*.public_ip}"]
}
Running output command from the task would then create the following pipeline variables
TF_OUT_LB_ADDRESS
TF_OUT_INSTANCE_IPS

## Out of Scope / Potential Follow-up Features
The following will not be supported in the initial changes but, may be added later.

### Export Only Specified Outputs
Export only outputs configured by the user to pipeline variables

### Export Sensitive Outputs
Terraform will redact any output that is marked as sensitive. Therefore, sensitive outputs will not be supported.

output "password" {
  sensitive = true
  value = ["${var.secret_password}"]
}
is emitted as

password = <sensitive>

**Comment by algattik** [comment]
Here's one way to do it https://github.com/microsoft/terraform-azure-devops-starter/blob/master/infrastructure/terraform-outputs-template.yml

**Comment by favoretti** [comment]
One challenge with this would be complex types in TF 0.12. We for instance output complete modules as outputs, which result in very complex arbitrary maps. 

Secrets in JSON output aren't redacted, but their `sensitive` property is set to `true`.

**Comment by charleszipp** [comment]
Started working on this today. In response to the challenge @favoretti with the complex maps/types. It seems just stringifying the value allows the value to be set fine as a string. Need to research if there is a length limit on this though. Alternatively, could just exclude object/tuple types and only set primitives.

```text
//BOOL
RAW:  	rg_name_bool bool true
VALUE:  rg_name_bool bool true

//NUMBER
RAW:  	rg_name_number number 1
VALUE:  rg_name_number number 1

//STRING
RAW:  	rg_name_string string A
VALUE:  rg_name_string string A

//OBJECT (stringify value)
RAW:  	rg_name_map [ 'object', { A: 'number', B: 'number', C: 'number' } ] { A: 1, B: 2, C: 3 }
VALUE:  rg_name_map [ 'object', { A: 'number', B: 'number', C: 'number' } ] 

//TUPLE (stringify value)
RAW:  	rg_name_tuple [ 'tuple', [ 'string', 'number', 'string' ] ] [ '1', 2, '3' ]
VALUE:  rg_name_tuple [ 'tuple', [ 'string', 'number', 'string' ] ] ["1",2,"3"]
```

**Comment by charleszipp** [comment]
@algattik @favoretti added with #135, would appreciate if you can take a look at the pr if you have time.
